Tribler/Test/test_tunnel_base.py

```diff
@@ -19,18 +21,34 @@ class TestTunnelBase(TestGuiAsServer):
 
     def startTest(self, callback, min_timeout=5, nr_relays=12, nr_exitnodes=4, crypto_enabled=True, bypass_dht=False):
 
+        print "\nHERE10"
         self.getStateDir()   # getStateDir copies the bootstrap file into the statedir
 
+        def print_proxy_setup_succsesful(self, exit_node, i):
+            if not nr_exitnodes:
+                print "Proxy %d setup successfully" % i
+            else:
+                print "Exit node %d setup successfully" % i
+
+
         def setup_proxies():
+            print "\nHERE12"
             tunnel_communities = []
             baseindex = 3
             for i in range(baseindex, baseindex + nr_relays):  # Normal relays
-                tunnel_communities.append(create_proxy(i, False, crypto_enabled))
+                self.CallConditional(60, create_proxy(i, False, crypto_enabled), print_proxy_setup_succsesful(False, i),
+                                 'Could not create normal proxy within 60 seconds')
+                # tunnel_communities.append(create_proxy(i, False, crypto_enabled))
```

Comment:
This section is where the test currently hangs because of a lack of proxies and/or exit nodes.